src/components/PostList/Post.js

```diff
@@ -22,6 +25,41 @@ const Post = ({post}) => {
     const postedBy = useSelector((state) =>
         state.allUsers.find((user) => user?._id === post.userId)
     );
+    const [liked, setLiked] = useState(post.liked.includes(currentUser?._id));
+    const [disliked, setDisliked] = useState(post.disliked.includes(currentUser?._id));
```

Comment:
issue is with these includes to setState -> fine initially but not on update.

Reply:
Yeah I haven't seen people use set state like this, not saying its impossible but I haven't tried it. I don't think you need to do uesState, you can just do 
```
const liked = post.liked.includes(currentUser?._id);
```
And then customize your own on liked function that will update the state.